src/components/animal/creation/AnimalCreationForm.vue

```diff
@@ -52,15 +54,8 @@ export default {
   emits: ['exit'],
   methods: {
     async createAnimal() {
-      const newAnimal = JSON.stringify(this.animal);
       try {
-        await fetch(`${process.env.VUE_APP_BASE_URL}/animals`, {
-          method: 'POST',
-          body: newAnimal,
-          headers: {
-            'content-type': 'application/json',
-          },
-        });
+        await createAnimal(this.animal);
```

Comment:
Il faut peut-être changer le nom de la méthode `createAnimal()` pour ne pas créer de confusion avec `createAnimal(this.animal)`.

Reply:
C'est vrai tu as des propositions pour ca ? On change l'api ou la méthode ?

Reply:
Pour moi ce serai bien de garder les noms sur l'api homogènes avec ceux du back et pour les proposiontions :
- `createAnimal()` -> `newAnimal()` ou `createAnimalFromAPI()`
- `deleteAnimal()` -> `removeAnimal()` ou `deleteAnimalFromAPI()`
- Du coup dans cette logique pour le PUT ce serait l'api à changer `saveAnima(this.animal)` -> `putAnimal(this.data.id, this.data)`

Reply:
Yep t'as raison, je pense que c'est le mieux.
Sinon quelques morceaux de doc pour implémenter des api sur vue :

- https://v3.vuejs.org/guide/composition-api-introduction.html#why-composition-api
- https://v3.vuejs.org/api/composition-api.html#setup

Reply:
Composition api a rien a voir, mais je comptais t'en parler a l'oral :)